pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -0,0 +1,115 @@
+package virtualmachine
+
+import (
+	"encoding/json"
+	"fmt"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+
+	ctlharvesterv1 "github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+func NewMutator(
+	setting ctlharvesterv1.SettingCache,
+) types.Mutator {
+	return &vmMutator{
+		setting: setting,
+	}
+}
+
+type vmMutator struct {
+	types.DefaultMutator
+	setting ctlharvesterv1.SettingCache
+}
+
+func (m *vmMutator) Resource() types.Resource {
+	return types.Resource{
+		Name:       "virtualmachines",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   kubevirtv1.SchemeGroupVersion.Group,
+		APIVersion: kubevirtv1.SchemeGroupVersion.Version,
+		ObjectType: &kubevirtv1.VirtualMachine{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+		},
+	}
+}
+
+func (m *vmMutator) Create(request *types.Request, newObj runtime.Object) (types.PatchOps, error) {
+	vm := newObj.(*kubevirtv1.VirtualMachine)
+	patchOps, err := m.patchResourceOvercommit(vm)
```

Comment:
Why do we patch VM resources when we are using kubevirt's overcommit feature?

Reply:
If we specify `resources.limits` on a VM, we'll lost the functionality of cluster-wide overcommit configuration of KubeVirt due to its implementation. See https://github.com/harvester/harvester/issues/1429#issuecomment-953031270

Reply:
The reason of updating KubeVirt overcommit config here is just a simple synchronization to prevent feature bugs.

Reply:
What does `feature bugs` mean and how can we prevent by setting these values? If we are not using the configuration I propose not touching it

Reply:
Sorry for the typo. The unused configuration part has been removed. Thanks.